src/BuiltInTools/AspNetCoreDeltaApplier/StartupHook.cs

```diff
@@ -28,6 +28,68 @@ public static void Initialize()
         });
     }
 
+    private static (List<Action<Type[]?>> BeforeUpdates, List<Action<Type[]?>> AfterUpdates) GetMetadataUpdateHandlerActions()
+    {
+        var beforeUpdates = new List<Action<Type[]?>>();
+        var afterUpdates = new List<Action<Type[]?>>();
+
+        foreach (Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
+        {
+            foreach (CustomAttributeData attr in assembly.GetCustomAttributesData())
+            {
+                if (attr.AttributeType.FullName != "System.Reflection.Metadata.MetadataUpdateHandlerAttribute")
+                {
+                    continue;
+                }
+
+                IList<CustomAttributeTypedArgument> ctorArgs = attr.ConstructorArguments;
+                if (ctorArgs.Count != 1 ||
+                    ctorArgs[0].Value is not Type handlerType)
+                {
+                    Log($"'{attr}' found with invalid arguments.");
+                    continue;
+                }
+
+                bool methodFound = false;
+
+                if (handlerType.GetMethod("BeforeUpdate", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static, new[] { typeof(Type[]) }) is MethodInfo beforeUpdate)
```

Comment:
It might be be useful to Log something if the name matches but the arg list is different.